stable_baselines3/her/her_replay_buffer.py

```diff
@@ -0,0 +1,293 @@
+from collections import deque
+from typing import Dict, List, Optional, Tuple, Union
+
+import numpy as np
+import torch as th
+from gym import spaces
+
+from stable_baselines3.common.buffers import BaseBuffer
+from stable_baselines3.common.type_aliases import ReplayBufferSamples
+from stable_baselines3.common.vec_env import VecNormalize
+from stable_baselines3.common.vec_env.obs_dict_wrapper import ObsDictWrapper
+from stable_baselines3.her.goal_selection_strategy import GoalSelectionStrategy
+
+
+class HerReplayBuffer(BaseBuffer):
+    """
+    Replay Buffer for sampling HER (Hindsight Experience Replay) transitions online.
+    These transitions will not be saved in the Buffer.
+
+    :param env: The training environment
+    :param buffer_size: The size of the buffer measured in transitions.
+    :param max_episode_length: The length of an episode. (time horizon)
+    :param goal_selection_strategy: Strategy for sampling goals for replay.
+        One of ['episode', 'final', 'future', 'random']
+    :param observation_space: Observation space
+    :param action_space: Action space
+    :param device: PyTorch device
+        to which the values will be converted
+    :param n_envs: Number of parallel environments
+    :her_ratio: (float) The ratio between HER replays and regular replays in percent (between 0 and 1, for online sampling)
+    """
+
+    def __init__(
+        self,
+        env: ObsDictWrapper,
+        buffer_size: int,
+        max_episode_length: int,
+        goal_selection_strategy: GoalSelectionStrategy,
+        observation_space: spaces.Space,
+        action_space: spaces.Space,
+        device: Union[th.device, str] = "cpu",
+        n_envs: int = 1,
+        her_ratio: float = 0.6,
```

Comment:
```suggestion
        her_ratio: float = 0.8,
```

to match the defaults